src/AvTranscoder/properties/StreamProperties.cpp

```diff
@@ -58,6 +59,99 @@ size_t StreamProperties::getStreamId() const
     return _formatContext->streams[_streamIndex]->id;
 }
 
+size_t StreamProperties::getBitRate() const
+{
+    if(!_codecContext)
+        throw std::runtime_error("unknown codec context");
+
+    // return bit rate of stream if present or VBR mode
+    if(_codecContext->bit_rate || _codecContext->rc_max_rate)
+        return _codecContext->bit_rate;
+
+    LOG_WARN("The bitrate of the stream '" << _streamIndex << "' of file '" << _formatContext->filename << "' is unknown.")
+
+    const AVMediaType streamType = getStreamType();
+
+    // compute audio bitrate
+    if(streamType == AVMEDIA_TYPE_AUDIO)
```

Comment:
This switch could be avoided making StreamProperties <code>virtual</code>, overridden by VideoProperties and AudioProperties, don't you think?